### DISCUSSION TITLE
APIMAN on SQL Server

### DISCUSSION THREAD

**Discussion by pmishra** [apiman/apiman#2635] [Help]
Dear All, I need some help with an issue I'm facing during an upgrade process. Four years ago, I set up APIMAN 1.5 on SQL Server, and everything was running smoothly until now. We're in the process of upgrading our environment: moving from Windows 2012 to 2019, JBoss 7.0 to WildFly 23.0.2.Final (specifically using Apiman WildFly Overlay 3.1.3.Final), and upgrading SQL Server from 2012 to 2019. Additionally, we're updating Java from version 8 to 17.

As part of this upgrade, I've performed a fresh installation of APIMAN 3.1.3 (the latest version) and I plan to migrate our APIs to the new installation using export and import procedures since it seemed to be the only viable option. I utilized the DDL script apiman_mssql15.ddl to set up the SQL Server database. However, I encountered an issue with the JSON data type in the notification and outbox tables, specifically with the (payload JSON NOT NULL) constraint. To work around this, I converted it to varchar(Max), and the database setup appeared to be successful.

Most of the software components were installed and configured without any hitches. However, I'm running into a problem with the deployment of apiman.war. The error message I'm receiving is as follows:

"{\"WFLYCTL0080: Failed services\" => {\"jboss.deployment.unit.\\\"apiman.war\\\".WeldStartService\" => \"Failed to start service
    Caused by: java.lang.reflect.GenericSignatureFormatError: Signature Parse error: Expected Field Type Signature
	Remaining input: javax.transaction.Transactional$TxType\"}}"

I have a few quick questions:

Will SQL Server not work at all with this upgrade?
Could the payload JSON NOT NULL constraint be causing this deployment issue?
What are the recommended next steps for me to tackle this problem?
I believe I'm almost there, but this issue is preventing me from making further progress. Any insights or suggestions would be greatly appreciated. Thanks in advance!
-Padam

**Comment by msavy** [comment]
WildFly 23 doesn't work on Java 17.

**Reply by pmishra** [reply]
It started working after using Java 11.

**Comment by msavy** [comment]
Sorry, I forgot to reply to your other questions

> Will SQL Server not work at all with this upgrade?

Not guaranteed to work.

> Could the payload JSON NOT NULL constraint be causing this deployment issue?

No

> What are the recommended next steps for me to tackle this problem?
> I believe I'm almost there, but this issue is preventing me from making further progress. Any insights or suggestions would be greatly appreciated. Thanks in advance!

I have supported builds for my customers that have been modified to enhance support for Microsoft SQL Server, but that's not going into the upstream.
